app/pages/project/index.cjsx

```diff
@@ -82,6 +87,20 @@ ProjectPage = React.createClass
       .then (pages) =>
         @setState {pages}
 
+  getSelectedWorkflow: (project, preferences) ->
+    @setState selectedWorkflow: 'PENDING'
+
+    preferredWorkflowID = preferences?.preferences.selected_workflow ? project.configuration?.default_workflow
+    if preferredWorkflowID?
+      apiClient.type('workflows').get preferredWorkflowID
```

Comment:
Yeah, it felt kinda weird to query since we've already got the ID. I think the `if workflow.active` check is more explicit. Does this affect the back end performance-wise at all?